### PR TITLE
Emit generator functions and yield expressions in ES6.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This is not the downlevel emit.  This simply emits the AST as-is for ES6 and above.  Fixes https://github.com/Microsoft/TypeScript/issues/2703

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -1097,6 +1097,7 @@ var __param = this.__param || function(index, decorator) { return function (targ
                                 default:
                                     return Comparison.LessThan;
                             }
+                        case SyntaxKind.YieldExpression:
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: Can you amend my comment? Also, did we start supporting spread?

**Comment by mhegazy** [comment]
:+1:

**Comment by rbuckton** [comment]
:+1:

**Comment by yuit** [comment]
:+1:

**Pull request merged**